test/functional/editions_controller_test.rb

```diff
@@ -200,4 +200,69 @@ class EditionsControllerTest < ActionController::TestCase
       end
     end
   end
+
+  context "#update" do
+    setup do
+      @answer = FactoryBot.create(:answer_edition)
+    end
+
+    should "show update and success message and render show template when saved" do
+      post :update, params: {
+        id: @answer.id,
+        edition: {
+          title: "The changed title",
+        },
+      }
+
+      assert_template "show"
+      assert_equal "Edition updated successfully.", flash[:success]
+      assert_equal "The changed title", AnswerEdition.last.title
+    end
+
+    should "show error message and render show template when title field is blank" do
+      post :update, params: {
+        id: @answer.id,
+        edition: {
+          title: "",
+        },
+      }
+
+      assert_template "show"
+      assert_select ".gem-c-error-summary__list-item", "Enter a title"
+    end
+
+    should "show error message and render show template when the edition could not be updated" do
+      Edition.any_instance.stubs(:save).raises(StandardError)
+      post :update, params: {
+        id: @answer.id,
+        edition: {
+          title: "A title",
+        },
+      }
+
+      assert_template "show"
+      assert_select ".gem-c-error-summary__list-item", "Due to a service problem, the edition couldn't be updated"
+    end
+
+    should "show error message and render show template when save returns false" do
```

Comment:
Sorry David, I think I was wrong when I said we needed this test—the "show error message and render show template when title field is blank" test is testing this case already.

I'm curious how this test is actually passing, since it _shouldn't_ be showing the service problem error message when save returns false (since there isn't an exception).

Reply:
OK, though I understood we were testing subtly different cases here, but I guess not if passing a blank string for title is the only way to make save return false. 

I am equally confused though as to how the test passes WRT to the "Service problem" error: this doesn't display in the browser if we call `@resource.save` rather than `@resource.save!` - which I think is what you'd expect. But if I add an assertion to the test for a blank title string that the "Service problem" error is *not* rendered we get a failure. 😕 

This is clearly wrong but I'm not sure how much to worry about it. It just feels a bit wrong to ignore the apparent contradiction here.

Reply:
It'd be _nice_ to have an understanding of why it's unexpectedly passing in case it's a symptom of something bigger being wrong, but also don't want to invest too much time into it if we're deleting the test…

Reply:
Yes, agreed. I've reverted the last commit that adds that test.